### DISCUSSION TITLE
Orthogonal dipoles modeling in cylindrical coordinates

### DISCUSSION THREAD

**Discussion by kalosu** [NanoComp/meep#2846] [Q&A]
Hello there again MEEP community, 

I have been using MEEP to model a single dipole emitter within a planar structure. 

From these simulations, I am able to obtain the electric field distributions in the near and far field via ```get_farfield``` evaluations. 

In this form, I am able to obtain the field at the far-field generated by a single dipole.

However, the type of emitter that I am interested in has two orthogonal dipoles which are perpendicular to some axis of symmetry (NV centers in nanodiamonds).

How could I model the radiation from these two orthogonal dipoles? More importantly, how would I obtain the total Poynting flux from the contribution of both dipoles? 

Also, how should I add up the obtained far-field components (Ex, Ey and Ez) for both orthogonal dipoles? Is there any reference available on how to perform these calculations?

Thanks a lot for the feedback and comments.

**Comment by stevengj** [comment]
Linearly polarized / orthogonal dipoles are equivalent to a sum or difference of two circularly polarized dipoles, in the same way that linearly polarized planewaves are a sum of circularly polarized planewaves: https://meep.readthedocs.io/en/latest/Scheme_Tutorials/Cylindrical_Coordinates/#scattering-cross-section-of-a-finite-dielectric-cylinder

So, in cylindrical coordinates you can just put in a circularly polarized dipole ($m=+1$), and then get the desired linear dipole response just by adding/subtracting the fields produced by the other circular polarization ($m=-1$).

(When adding up total powers, often you don't even need this — it is equivalent to just look at $m = \pm 1$, since $m = +1$ and $m = -1$ fields are orthogonal when integrated over $\phi$.)

**Reply by kalosu** [reply]
Hi @stevengj ,

Thanks a lot for the feedback.
If I understood it correctly, I can obtain the desired polarization response by changing the way I combine the $m=+1$ and $m=-1$ field terms right?

For the "ex" polarization (a dipole polarized along r in the rz plane in cylindrical coordinates), I use $m_{p1} + m_{m1}$ and for the perpendicular polarization I would use $m_{p1} - m_{m1}$. Is this correct?

Using these linear combinations, I can now obtain two sets of 6 field components (Ex, Ey, Ez, Hx, Hy,Hz) in the far-field. Each set corresponding to the "ex" polarization and the perpendicular one.

At the end I am interested in obtaining the combined total far-field radiation characteristics from both perpendicular dipoles. This would be equivalent to the intensity distribution from an unpolarized dipole right? 

I usually obtain the Poynting vector components from the field components and then estimate the projection along the surface of a half-sphere, i.e, the intensity normal to the surface of the sphere.

How could I obtain the total radiation profile product of both perpendicular dipoles? Should I just add up the individual contributions from both dipoles? (i.e, simple pointwise summation of the estimated intensity values from the Poynting vector components)

Thanks for your comments and insights

**Reply by kalosu** [reply]
Hi again @stevengj,

I have been reading a bit more about the specific type of emitter that I am interested in modeling.

To be more precise, I want to model the emission from NV centers in nanodiamonds which are embedded in a planarized structure.

From many publications I can see that the radiation from these NV centers is described by two dipoles which are orthogonal to each other and that are perpendicular to the axis of symmetry of the nanocrystal that contains the NV center defect. Here is an example publication in where this idea is further explained:
https://opg.optica.org/oe/fulltext.cfm?uri=oe-22-4-4379&id=279598

Now, if I want to bring this "double dipole" modeling approach to a MEEP simulation, I suppose I can use what you suggested on using linear combinations of $m=-1$ and $m=+1$ circularly polarized dipoles depending on the desired polarization that I want for a single dipole. (addition for a dipole polarized in the rz plane and substraction for a dipole polarized perpendicular to the rz plane in a cylindrical coordinates simulation)

However, in this case I would have two sets of field distributions right? If I am interested in the total intensity I can sum up incoherently the Poynting vector components right? 

But what if I am also interested in the "total field components". From the paper I shared here, I understand that these orthogonal dipoles do not represent two independent radiated "photons". These orthogonal dipoles are used to represent any desired dipole polarization from which a single set of electromagnetic field components arise as the product of the interaction of a single dipole possesing the "desired dipole polarization" and the environment. 

What should I do with my MEEP field components to reach the same idea of being able to have a single set of total field components from a dipole that has any "desired dipole polarization"? (given that I have the field components associated to the  $m=-1$ and $m=+1$  sources)

Do you have some comments about this?
